File: Source/UI/src/app/models/Restaurante.ts
import { Usuario } from "./Usuario";
import { Comentario } from "./Comentario";

export class Restaurante {
    id: string;
    nome: string;
    endereco: string;
    votos: number;
    usuariosVotantes: Array<Usuario>;
    comentarios: Array<Comentario>;
    escolhido: boolean = false;
    dataUltimaEscolha: Date | null;
    votadoNaSemana: boolean;
    
    constructor(nome: string, endereco: string, votos: number, usuariosVotantes: Usuario[], comentarios: Comentario[]) {
        this.nome = nome;
        this.endereco = endereco;
        this.votos = votos;
        this.usuariosVotantes = usuariosVotantes;
        this.comentarios = comentarios;
    }
}

File: Source/Domain/Restaurante.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace HoraSagradaWebApi.Domain
{
    public class Restaurante
    {
        [Key]
        public Guid Id { get; set; }
        [Required]
        [StringLength(150)]
        public string Nome { get; set; }
        public string Endereco { get; set; }
        public int? Votos { get; set; }
        public IEnumerable<Usuario> UsuariosVotantes { get; set; }
        public bool? Escolhido { get; set; }
        public DateTime? DataUltimaEscolha { get; set; }
        public bool? VotadoNaSemana { get; set; }
    }
}